table.go

```diff
@@ -663,22 +684,55 @@ func (t *Table) appender() (*TableBlock, func(), error) {
 			return nil, nil, err
 		}
 
-		if block.Size() < t.db.columnStore.activeMemorySize {
+		blockSize := block.Size()
+		if t.db.columnStore.snapshotTriggerSize != 0 &&
+			blockSize-block.lastSnapshotSize.Load() > t.db.columnStore.snapshotTriggerSize {
```

Comment:
The only place where I'd like some more context in form of a comment.
Why `blockSize - block.lastSnapshotSize`?

Reply:
Thanks for the comment. That definitely helps. I'm still wondering:
Why does the `block.lastSnapshotSize` need to be subtracted? Shouldn't `blockSize > t.db.columnStore.snapshotTriggerSize` be enough here? Why does the last snapshot size matter?

Reply:
Because we want to get the number of "new" bytes since the last snapshot. Otherwise we'll be snapshotting every new write once the block size exceeds the trigger size (e.g. if the trigger size is `10`, we'll snapshot at `blockSize=10,11,12,13,14 etc...` when we only want to snapshot at `blockSize=10,20,30,40 etc...`)